### DISCUSSION TITLE
UIButton：How to change the background color of a button to an custom image, and how to change the window background to transparent?

### DISCUSSION THREAD

**Discussion by jike2018** [StereoKit/StereoKit#923] [Q&A]
![Snipaste_2024-03-26_14-31-29](https://github.com/StereoKit/StereoKit/assets/48623339/de749fea-b20c-416f-844f-f110f1dd9d6f)

**Comment by maluoi** [comment]
To make a window "transparent", you can either use `UI.WindowBegin`/`End` with `UIWin.Head` or `UIWin.Empty`, or skip the window and use `UI.Push`/`PopSurface`.

Making a button background use an image is more work. I'd probably recommend avoiding it unless it's critical? You basically need to texture the button in order to do it, which gets into some technical art work. If it's something where you have no choice in the matter, or you're comfortable with that sort of work, I'd peek here at an ["Edge decoration" sample](https://gist.github.com/maluoi/de969733c0016d5ee2e6afb3cb5c6c66) I put together. This sample is more targeted towards solid colors, but totally works with images as well.

You may also be interested in creating your own custom buttons as well. If you use `UI.ButtonBehavior`, you can get all of SK's button behavior, but without any of the rendering. This lets you render whatever you'd like instead of what SK provides.
